way_to_home/custom_user/views.py

```diff
@@ -58,7 +59,10 @@ def registration_confirm(request, token):
         return HttpResponse('received email is not valid', status=400)
 
     is_updated = user.update(is_active=True)
-    if not is_updated:
+
+    profile = UserProfile.create(user)
+
+    if not (is_updated or profile):
         return HttpResponse('database operations is failed', status=400)
```

Comment:
Can you imagine the situation if the user profile is created and the user is not activated. When the next activation attempt, will you create another profile? I think you need to put it in the transaction.

Reply:
changed that by moving profile creation after the activation check